pkg/master/master_openapi_test.go

```diff
@@ -34,7 +34,7 @@ import (
 	openapinamer "k8s.io/apiserver/pkg/endpoints/openapi"
 	genericapiserver "k8s.io/apiserver/pkg/server"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
-	openapigen "k8s.io/kubernetes/pkg/generated/openapi"
+	generatedopenapi "k8s.io/apiextensions-apiserver/pkg/generated/openapi"
```

Comment:
why is it correct to start testing `k8s.io/apiextensions-apiserver/pkg/generated/openapi` and stop testing `openapigen "k8s.io/kubernetes/pkg/generated/openapi"`?

I would have expected an added test to cover the new package (and to put that test within `k8s.io/apiextensions-apiserver`)

Reply:
```
find . -name openapi | grep generated
./staging/src/k8s.io/apiextensions-apiserver/pkg/generated/openapi
./staging/src/k8s.io/sample-apiserver/pkg/generated/openapi
./pkg/generated/openapi
```
When I look under ./pkg/generated/openapi :
```
grep GetOpenAPIDefinitions ./pkg/generated/openapi/*
./pkg/generated/openapi/openapi_test.go:	for name, value := range generatedopenapi.GetOpenAPIDefinitions(dummyRef) {
```
I wonder where the generated source code for GetOpenAPIDefinitions of "k8s.io/kubernetes/pkg/generated/openapi" is located.

Reply:
I found the following definitions of GetOpenAPIDefinitions func:
```
staging/src/k8s.io/apiextensions-apiserver/pkg/generated/openapi/zz_generated.openapi.go:func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
staging/src/k8s.io/code-generator/_examples/apiserver/openapi/zz_generated.openapi.go:func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
staging/src/k8s.io/sample-apiserver/pkg/generated/openapi/zz_generated.openapi.go:func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
```
The first seems to be the only choice for the underlying test.

Reply:
Assuming the first of above should be referenced, I need to get over this:
```
W0109 18:06:45.680] .-> //vendor/k8s.io/apiextensions-apiserver/pkg/generated/openapi:go_default_library
W0109 18:06:45.680] |   //vendor/k8s.io/apiextensions-apiserver/pkg/generated/openapi:zz_generated.openapi.go
W0109 18:06:45.681] |   //vendor/k8s.io/apiextensions-apiserver/pkg/generated/openapi:zz_generated.openapi
W0109 18:06:45.681] `-- //vendor/k8s.io/apiextensions-apiserver/pkg/generated/openapi:go_default_library
W0109 18:06:45.681] This cycle occurred because of a configuration option
```